src/Explorer/Notebook/NotebookContainerClient.ts

```diff
@@ -131,21 +131,29 @@ export class NotebookContainerClient {
       } else if (response.status === HttpStatusCodes.NotFound) {
         throw new AbortError(response.statusText);
       }
-      throw new Error();
+      throw new Error(response.statusText);
     } else {
       return undefined;
     }
   }
 
   private checkStatus(): boolean {
-    if (NotebookUtil.isPhoenixEnabled()) {
-      if (useNotebook.getState().containerStatus?.status === Constants.ContainerStatusType.Disconnected) {
+    if (useNotebook.getState().isPhoenix) {
```

Comment:
check when you should return true and do only that check, the resetting of other states should be done based on the health check, not the memory call

Reply:
Remove setting/resetting state in scheduleHeartbeat as well